geoh5py/workspace/workspace.py

```diff
@@ -678,50 +574,168 @@ def find_type(
+        for key, val in self._groups.items():
+            entity = val.__call__()
+            if entity is not None:
+                groups_name[key] = entity.name
+        return groups_name
 
-    def find_entity(self, entity_uid: uuid.UUID) -> Optional["Entity"]:
-        """Get all active entities registered in the workspace.
+    @property
+    def list_objects_name(self) -> Dict[uuid.UUID, str]:
         """
-        return (
-            self.find_group(entity_uid)
-            or self.find_data(entity_uid)
-            or self.find_object(entity_uid)
-        )
+        :obj:`dict` of UUID keys and name values for all registered Objects
+        """
+        objects_name = {}
+        for key, val in self._objects.items():
+            entity = val.__call__()
+            if entity is not None:
+                objects_name[key] = entity.name
+        return objects_name
+
+    @property
+    def modified_attributes(self) -> bool:
+        """
+        :obj:`list[str]` List of attributes to be updated in associated workspace
+        :obj:`~geoh5py.workspace.workspace.Workspace.h5file`.
+        """
+        return self._modified_attributes
+
+    @modified_attributes.setter
+    def modified_attributes(self, value: bool):
+        self._modified_attributes = value
+
+    @property
+    def name(self) -> str:
+        """
+         :obj:`str` Name of the project
+        """
+        return self._name
+
+    def _register_type(self, entity_type: "EntityType"):
+        weakref_utils.insert_once(self._types, entity_type.uid, entity_type)
+
+    def _register_group(self, group: "group.Group"):
+        weakref_utils.insert_once(self._groups, group.uid, group)
+
+    def _register_data(self, data_obj: "data.Data"):
+        weakref_utils.insert_once(self._data, data_obj.uid, data_obj)
+
+    def _register_object(self, obj: "object_base.ObjectBase"):
+        weakref_utils.insert_once(self._objects, obj.uid, obj)
+
+    @property
+    def root(self) -> Optional[Entity]:
+        """
+        :obj:`~geoh5py.groups.root_group.RootGroup` entity.
+        """
+        return self._root
+
+    @root.setter
+    def root(self, entity) -> Optional[Entity]:
+        if self._root is None:
+            assert isinstance(
+                entity, RootGroup
+            ), f"The given root entity must be of type {RootGroup}"
+            self._root = entity
+
+        return self._root
+
+    @staticmethod
+    def save_entity(entity: Entity, close_file: bool = True, add_children: bool = True):
+        """
+        Save or update an entity to geoh5
+
+        :param entity: Entity to be written to geoh5
+        :param close_file: Close the geoh5 database after writing is completed
+        :param add_children: Add children entities to geoh5
+        """
+        H5Writer.save_entity(entity, close_file=close_file, add_children=add_children)
+
+    @property
+    def version(self) -> float:
+        """
+        :obj:`float` Project version
```

Comment:
wrong doc and missing period
```suggestion
        :obj:`float` Version of the geoh5 file format.
```